packages/react-client/src/ReactFlightClient.js

```diff
@@ -515,37 +518,40 @@ function initializeModelChunk<T>(chunk: ResolvedModelChunk<T>): void {
 
   const resolvedModel = chunk.value;
 
-  // We go to the CYCLIC state until we've fully resolved this.
+  // We go to the BLOCKED state until we've fully resolved this.
   // We do this before parsing in case we try to initialize the same chunk
   // while parsing the model. Such as in a cyclic reference.
-  const cyclicChunk: CyclicChunk<T> = (chunk: any);
-  cyclicChunk.status = CYCLIC;
+  const cyclicChunk: BlockedChunk<T> = (chunk: any);
```

Comment:
Yea I was considering but decided to keep it. It's not a separate status but it's still cyclic at this point which is why we feel comfortable invoke the listeners later in this function - because nobody should be listening to it (not 100% true because a `require()` into a module could but not likely).